### PR TITLE
Law: time-independent Schrödinger equation

### PR DESCRIPTION
**Pull request by alesanter**

### CODE REVIEW

File: symplyphysics/laws/quantum_mechanics/schrodinger/time_independent_equation.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+# Law: (-hbar**2 / (2 * m)) * d^2(psi(x))/dx^2 + U(x) * psi(x) = E * psi(x)
+## psi(x) - wavefunction of particle
+## x - position
+## hbar - reduced Planck constant
+## m - particle mass
+## U(x) - potential energy
+## E - energy of particle, eigenvalue of the equation
+
+wavefunction = Function("wavefunction", 1 / sqrt(units.length), real=1)
```
Comment (blackyblack) [inline symplyphysics/laws/quantum_mechanics/schrodinger/time_independent_equation.py:23]: Use True instead of 1

File: symplyphysics/laws/quantum_mechanics/schrodinger/time_independent_equation.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+position = Symbol("position", units.length, real=1)
+potential_energy = Function("potential_energy", units.energy, real=1)
+particle_mass = clone_symbol(symbols.basic.mass, "particle^mass", positive=1)
```
Comment (blackyblack) [inline symplyphysics/laws/quantum_mechanics/schrodinger/time_independent_equation.py:26]:
```suggestion
particle_mass = clone_symbol(symbols.basic.mass, "particle_mass", positive=1)
```

File: symplyphysics/laws/quantum_mechanics/schrodinger/time_independent_equation.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    clone_symbol,
+)
+
+# Description
+## The Schrödinger equation is a linear partial differential equation that governs the wave
+## function of a quantum-mechanical system.
+
+# Law: (-hbar**2 / (2 * m)) * d^2(psi(x))/dx^2 + U(x) * psi(x) = E * psi(x)
+## psi(x) - wavefunction of particle
+## x - position
```
Comment (blackyblack) [inline symplyphysics/laws/quantum_mechanics/schrodinger/time_independent_equation.py:17]: Does this law have 2 or 3 dimensional form?

File: symplyphysics/laws/quantum_mechanics/schrodinger/time_independent_equation.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+particle_energy = Symbol("particle_energy", units.energy, positive=1)
+
+law = Eq(
+    (-1 * hbar**2 / (2 * particle_mass)) * Derivative(wavefunction(position), position, 2)
+    + potential_energy(position) * wavefunction(position),
+    particle_energy * wavefunction(position),
+)
+
+# The solutions of the Schrödinger equation vary drastically with the kind of potential energy
+# function is used and depend on the boundary conditions.
```
Comment (blackyblack) [inline symplyphysics/laws/quantum_mechanics/schrodinger/time_independent_equation.py:36]: We need some test. Could you look for any sample data that can be fed into this equation and verify with expected result?

File: symplyphysics/laws/quantum_mechanics/schrodinger/time_independent_equation.py (hunk 3 above)
Comment (alesanter) [reply symplyphysics/laws/quantum_mechanics/schrodinger/time_independent_equation.py:17]: Yes. I'll mention the dimension in the file name

File: symplyphysics/laws/quantum_mechanics/schrodinger/time_independent_equation.py (hunk 4 above)
Comment (alesanter) [reply symplyphysics/laws/quantum_mechanics/schrodinger/time_independent_equation.py]: I'm not sure what can be calculated here. The wave function itself? We would need to at least specify the potential function to be able to somehow solve the equation, but that could be done in separate laws (free particle, potential well, harmonic oscillator, etc).

Maybe letting the user provide their own wave function expression and checking if it satisfies the equation will do it.

Comment (blackyblack) [reply symplyphysics/laws/quantum_mechanics/schrodinger/time_independent_equation.py]: We do not need `calculate` to write a test. We need some reference inputs and output.

**Review by blackyblack — APPROVED** [review]
